torch/_prims/nvfuser_executor.py

```diff
@@ -209,18 +229,35 @@ def maybe_partition_graph(gm: GraphModule):
             if node.op == "call_module" and "fused_" in node.name:
                 nvfuser_submodule = getattr(partitioned_graph, node.name)
                 partitioned_graph.delete_submodule(node.target)
-                gm.add_submodule(node.target, NvfuserGraphModule(nvfuser_submodule))
+                gm.add_submodule(
+                    node.target,
+                    NvfuserGraphModule(nvfuser_submodule, use_python_fusion_cache),
+                )
 
         return partitioned_graph, any_unsupported
     else:
         return gm, any_unsupported
 
 
-def nvfuser_execute_partitioned(gm: GraphModule, *args):
+def nvfuser_execute_partitioned(gm: GraphModule, *args, executor_parameters=None):
+    executor_parameters = executor_parameters or DEFAULT_NVFUSER_PYTHON_CONFIG
+    # maybe_partition_graph function is cached so we can't use non-hashable arguments
+    allow_single_op_fusion = executor_parameters.get(
+        "allow_single_op_fusion",
+        DEFAULT_NVFUSER_PYTHON_CONFIG["allow_single_op_fusion"],
```

Comment:
why do we need a default value here? wasn't `executor_parameters` already a union with the default?

Reply:
If someone passes an empty dictionary or a dictionary with parameters omitting `allow_single_op_fusion` we'd still use the default and not error out. Calling `torch._prims.executor.execute(graph_module, args, executor_parameters={"none": None})` is allowed.
I should add tests for that.

`executor_parameters` is not a union with the default here, should it be? If `executor_parameters` is not specified or None then `executor_parameters==DEFAULT_NVFUSER_PYTHON_CONFIG`, if it's not None then `executor_parameters` is whatever was passed in by the user.

Reply:
"executor_parameters = executor_parameters or DEFAULT_NVFUSER_PYTHON_CONFIG"

Sorry I thought `or` here is a union `|`... :facepalm: